File: dart/lib/model/v3_disruption_route.dart
part of swagger.api;

class V3DisruptionRoute {
  /* Transport mode identifier */
  int routeType = null;
/* Route identifier */
  int routeId = null;
/* Name of route */
  String routeName = null;
/* Route number presented to public (i.e. not route_id) */
  String routeNumber = null;
/* GTFS Identifer of the route */
  String routeGtfsId = null;

  V3DisruptionDirection direction = null;

  V3DisruptionRoute();

  @override
  String toString() {
    return 'V3DisruptionRoute[routeType=$routeType, routeId=$routeId, routeName=$routeName, routeNumber=$routeNumber, routeGtfsId=$routeGtfsId, direction=$direction, ]';
  }

  V3DisruptionRoute.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    routeType = json['route_type'];
    routeId = json['route_id'];
    routeName = json['route_name'];
    routeNumber = json['route_number'];
    routeGtfsId = json['route_gtfs_id'];
    direction = new V3DisruptionDirection.fromJson(json['direction']);
  }

  Map<String, dynamic> toJson() {
    return {
      'route_type': routeType,
      'route_id': routeId,
      'route_name': routeName,
      'route_number': routeNumber,
      'route_gtfs_id': routeGtfsId,
      'direction': direction
     };
  }

  static List<V3DisruptionRoute> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3DisruptionRoute>() : json.map((value) => new V3DisruptionRoute.fromJson(value)).toList();
  }

  static Map<String, V3DisruptionRoute> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3DisruptionRoute>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3DisruptionRoute.fromJson(value));
    }
    return map;
  }
}

File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3DisruptionRoute.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3DisruptionRoute {
    /// <summary>
    /// Transport mode identifier
    /// </summary>
    /// <value>Transport mode identifier</value>
    [DataMember(Name="route_type", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "route_type")]
    public int? RouteType { get; set; }

    /// <summary>
    /// Route identifier
    /// </summary>
    /// <value>Route identifier</value>
    [DataMember(Name="route_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "route_id")]
    public int? RouteId { get; set; }

    /// <summary>
    /// Name of route
    /// </summary>
    /// <value>Name of route</value>
    [DataMember(Name="route_name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "route_name")]
    public string RouteName { get; set; }

    /// <summary>
    /// Route number presented to public (i.e. not route_id)
    /// </summary>
    /// <value>Route number presented to public (i.e. not route_id)</value>
    [DataMember(Name="route_number", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "route_number")]
    public string RouteNumber { get; set; }

    /// <summary>
    /// GTFS Identifer of the route
    /// </summary>
    /// <value>GTFS Identifer of the route</value>
    [DataMember(Name="route_gtfs_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "route_gtfs_id")]
    public string RouteGtfsId { get; set; }

    /// <summary>
    /// Gets or Sets Direction
    /// </summary>
    [DataMember(Name="direction", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "direction")]
    public V3DisruptionDirection Direction { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3DisruptionRoute {\n");
      sb.Append("  RouteType: ").Append(RouteType).Append("\n");
      sb.Append("  RouteId: ").Append(RouteId).Append("\n");
      sb.Append("  RouteName: ").Append(RouteName).Append("\n");
      sb.Append("  RouteNumber: ").Append(RouteNumber).Append("\n");
      sb.Append("  RouteGtfsId: ").Append(RouteGtfsId).Append("\n");
      sb.Append("  Direction: ").Append(Direction).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
